cmd/genbindings/clang2il.go

```diff
@@ -165,6 +165,29 @@ nextTopLevel:
 	return &ret, nil // done
 }
 
+// shouldPreferQualType returns true if we should use the qualType instead of
+// the desugared type based on certain type patterns
+func shouldPreferQualType(qualType string) bool {
+	if strings.Contains(qualType, "intptr") || strings.Contains(qualType, "_t") ||
+		strings.HasPrefix(qualType, "uint") || strings.Contains(qualType, "ushort") ||
+		strings.Contains(qualType, "quint") || strings.Contains(qualType, "qint") ||
+		strings.Contains(qualType, "qptrdiff") || strings.HasPrefix(qualType, "qsize") {
```

Comment:
@arnetheduck Thoughts? 😅 

This diff is still showing more changes than it should so I'm looking into why that is.

Reply:
This is why. 😔 

```
DESUGARED: long long
QUALTYPE: QIntegerForSize<8>::Signed
DESUGARED: unsigned long long
QUALTYPE: QIntegerForSize<8>::Unsigned
DESUGARED: unsigned long long
QUALTYPE: QIntegerForSizeof<void *>::Unsigned
DESUGARED: long long
QUALTYPE: QIntegerForSizeof<void *>::Signed
```

Reply:
There is a list of int types already in intermediate.go IntType() that may be useful, it could be put in a common function.

There are some places in genbindings where we do need to accurately know the type before typedefs are applied, in order to (e.g.) correctly hit a precice overload of a function. But if clang itself is considering these types equivalent, then it shouldn't cause any problems.